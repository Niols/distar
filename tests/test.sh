###########################################
#             Test - distar               #
#           Copyright (C) 2018            #
#   Yann Régis-Gianas - Étienne Marais    #
###########################################

#!/bin/sh

# Functions for printing
print_error () {
    echo -e "- \033[1;31mFAILED\033[0m $1"
}

print_ok () {
    echo -e "- \033[1;32mOK\033[0m $1"
}

# Functions to calculate stats
succes=0
failure=0
total=0

succeed () {
    success=$((succes+1))
    total=$((total+1))
}

failed () {
    failure=$((failure+1))
    total=$((total+1))
}


# Roam all the directories to launch test.sh
echo -e "\n=== Situational tests ===\n"
for dir in */
do
     echo -e "--> \033[1;39m$dir\033[0m"

     # Launch local test 
     sh $dir/test.sh

     # Count sucess and failure
     if [ $? -eq 0 ];then
         succeed
         print_ok 
     else
         failed
         print_error 
     fi
     echo -e #Add one line between tests
done
echo "---------"
echo "$success succeed, $failure failed, total : $total"

