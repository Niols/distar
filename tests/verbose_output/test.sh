#!/bin/sh

set -eu
cd "$(dirname "$0")"

# Launch distar in verbose mode and store output in a file
../distar -v source_1.ml source_2.ml target.md > test.output

# Keep diff output to compare in case of error
output=$(diff test.output test.expected)

# Check diff exit code
if [ $? -eq 0 ]
then
    exit 0
else
    echo $output
    exit 1
fi
