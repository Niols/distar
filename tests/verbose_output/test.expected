distar
|-source(s): source_1.ml / source_2.ml
|-target: target.md
